src/FactoryGirl.js

```diff
@@ -121,13 +122,17 @@ export default class FactoryGirl {
   }
 
   cleanUp() {
-    const promises = [];
-    for (const [adapter, model] of this.created) {
-      promises.push(adapter.destroy(model, model.constructor));
+    const createdArray = [];
+    for (const c of this.created) {
+      createdArray.push(c);
     }
+    const promise = bluebird.each(
+      createdArray,
+      ([adapter, model]) => adapter.destroy(model, model.constructor)
+    );
     this.created.clear();
     this.resetSeq();
-    return Promise.all(promises);
+    return promise;
```

Comment:
I'd like to avoid making bluebird a runtime dependency. You can achieve the same as follows:

```js
const promise = createdArray.reduce((prev, [adapter, model]) => {
  prev.then(() => adapter.destroy(model, model.constructor)
}, Promise.resolve())
```